docs/pages/enroll-resources/database-access/enroll-self-hosted-databases/oracle-exadata.mdx

```diff
@@ -0,0 +1,428 @@
+| `--roles`    | List of roles to assign to the user. The builtin `access` role allows them to connect to any database server registered with Teleport.   |
+| `--db-users` | List of database usernames the user will be allowed to use when connecting to the databases. A wildcard allows any user.                 |
+| `--db-names` | List of logical databases (aka schemas) the user will be allowed to connect to within a database server. A wildcard allows any database. |
+
+<Admonition type="warning">
+  Database names are only enforced for PostgreSQL and MongoDB databases.
+</Admonition>
+
+For more detailed information about database access controls and how to restrict
+access see [RBAC](../../database-access/rbac.mdx) documentation.
+
+## Step 2/6. Configure Oracle Exadata
+
+<Admonition type="note">
+
+This guide assumes default configuration of the Oracle Exadata system. In particular:
+- Preconfigured TCPS listener.
+- An existing database certificate in the grid wallet.
+- A single Oracle Exadata VM is accessible to the `opc` user through ssh with a hostname `demodb-vm`. Repeat the steps to configure additional VMs.
+- The SCAN DNS name is `demodb-vm-scan.exadatadomain.oke.oraclevcn.com`
+
+Adjust the commands to match your configuration.
+
+</Admonition>
+
+Teleport uses mutual TLS authentication with self-hosted databases. These databases must be configured with Teleport's certificate authority to be able to verify client certificates.
+
+Login to Teleport on your machine.
+
+```code
+$ tsh login --proxy=<Var name="example.teleport.sh" /> --user=alice
+```
+
+Export the database client CA and copy it to target machine.
+
+```code
+# Export Teleport's database client certificate authority
+$ tctl auth export --type=db-client > teleport-db-client-ca.crt
+# Copy the CA to the Oracle Exadata VM
+$ scp teleport-db-client-ca.crt opc@demodb-vm:/tmp/teleport-db-client-ca.crt
+```
+
+Login to the target machine and switch to `grid` user.
+
+```code
+# Connect to the VM
+$ ssh opc@demodb-vm
+# Switch to grid user
+$ sudo su - grid
+```
+
+Update the TCPS wallet to trust the Teleport User Database CA.
+
+```code
+# Read the wallet password
+$ mkstore -wrl /u01/app/oracle/admin/cprops/cprops_wallet -nologo -viewEntry grid_tcps_wallet_passwd
+grid_tcps_wallet_passwd = <wallet password>
+# Update the grid TCPS wallet; provide the password when prompted
+$ orapki wallet add -wallet "/var/opt/oracle/dbaas_acfs/grid/tcps_wallets" -trusted_cert -cert /tmp/teleport-db-client-ca.crt
+```
+
+Enable TLS auth for grid listener by editing the configuration.
+
+```code
+# Edit listener.ora to read
+# SSL_CLIENT_AUTHENTICATION = TRUE
+$ vi $ORACLE_HOME/network/admin/listener.ora
+...
+```
+
+For each database, update the configuration in the database home. Assuming a single database called `spark`:
+
+```code
+# Switch to oracle user
+$ sudo su - oracle
+# Load env file for database 'spark'
+$ . spark.env 
```

Comment:
I understand "spark" is just an example database you're using but it's not entirely clear whether this `.env` file is supposed to always exist and where?

Reply:
It is automatically created for each database. If you have ten databases, you will find ten different `.env` files in the home directory of Oracle user.

Reply:
I've rephrased this section to explain what is an example and what are expectations.